ibis/expr/types/rich.py

```diff
@@ -0,0 +1,78 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from ibis.expr import types as ir
+
+if TYPE_CHECKING:
+    from collections.abc import Iterable
+
+    import rich.panel
+    import rich.table
+    from rich.console import RenderableType
+
+
+try:
+    from rich.jupyter import JupyterMixin
+except ImportError:
+
+    class FixedTextJupyterMixin:
+        """No-op when rich is not installed."""
+else:
+
+    class FixedTextJupyterMixin(JupyterMixin):
+        """JupyterMixin adds a spurious newline to text, this fixes the issue."""
+
+        def _repr_mimebundle_(
+            self, include: Iterable[str], exclude: Iterable[str], **kwargs
+        ):
+            exclude = [*exclude, "text/plain"]
+            try:
+                bundle = super()._repr_mimebundle_(include, exclude, **kwargs)
+            except Exception:  # noqa: BLE001
+                return None
+            else:
+                bundle["text/plain"] = capture_rich_renderable(self, no_color=False)
```

Comment:
This is a change in this PR. Before, we just took the output of `super()._repr_mimebundle_()` and stripped the bogus newline. But now, I wanted to unify our codepaths. Now, both of `__repr__` and `_repr_mimebundle_` go through `capture_rich_renderable` so they should have the same behavior and there is only one place to modify it. We still rely on `super()._repr_mimebundle_(include, exclude, **kwargs)` to give us the html representation.